### PR TITLE
[Docs] Clarify Storage mounting details

### PR DESCRIPTION
**Pull request by romilbhardwaj**
Fixes some incorrect statements about storage mounting in our perf benchmark.

### CODE REVIEW

**Comment by concretevitamin** [comment]
Thanks @romilbhardwaj. Shall we also add this requested info to the docs page?  Will be easier to find than this markdown.

**Comment by romilbhardwaj** [comment]
Added!

File: docs/source/reference/storage.rst [hunk 1]
```diff
@@ -63,8 +63,8 @@ When specifying a storage object, you can specify either of two modes:
     they are accessed by applications. This mode also allows applications to
     write to the mount path. All writes are replicated to remote bucket (and
     any other VMs mounting the same bucket). Please note that this mode
-    uses a close-to-open consistency model, which means a file write is
-    committed to the backing store only after :code:`close()` is called on it.
+    uses a close-to-open consistency model. This means calling :code:`close()`
+    on a file will guarantee that any subsequent reads will see the latest data.
```
Comment (concretevitamin) [inline docs/source/reference/storage.rst:67]: “close-to-open” can be unfamiliar. Maybe we can add a sentence using the user’s own words, along the lines of “every close will trigger an re-upload of the entire file (true?) to the backing bucket”.

**Review by concretevitamin — APPROVED** [review]
Thanks!

File: docs/source/reference/storage.rst (hunk 1 above)
Comment (romilbhardwaj) [reply docs/source/reference/storage.rst]: Thanks! I added this but it was making the bullet point too long, so I moved these details to two notes after the example. PTAL

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks @romilbhardwaj.

**Pull request merged**